subsys/bluetooth/audio/pacs.c

```diff
@@ -666,92 +758,254 @@ static int pac_notify_snk(void)
+	if (err != 0) {
+		LOG_ERR("Failed to get conn info: %d", err);
+		return;
+	}
+	if (info.state != BT_CONN_STATE_CONNECTED) {
+		/* Not connected */
+		return;
 	}
 
-	return 0;
+	k_work_submit(&deferred_nfy_work);
 }
 
-static int supported_contexts_notify(void)
+static void pacs_auth_pairing_complete(struct bt_conn *conn, bool bonded)
 {
-	struct bt_pacs_context context = {
-		.snk = sys_cpu_to_le16(snk_supported_contexts),
-		.src = sys_cpu_to_le16(src_supported_contexts),
-	};
-	int err;
+	LOG_DBG("%s paired (%sbonded)", bt_addr_le_str(bt_conn_get_dst(conn)),
+		bonded ? "" : "not ");
 
-	err = bt_gatt_notify_uuid(NULL, BT_UUID_PACS_SUPPORTED_CONTEXT, pacs_svc.attrs,
-				  &context, sizeof(context));
-	if (err != 0 && err != -ENOTCONN) {
-		LOG_WRN("Supported Audio Contexts notify failed: %d", err);
+	if (!bonded) {
+		return;
+	}
 
-		return err;
+	/* Check if already in list, and do nothing if it is */
+	for (uint8_t i = 0; i < ARRAY_SIZE(clients); i++) {
+		if (atomic_test_bit(clients[i].flags, FLAG_ACTIVE) &&
+		    bt_addr_le_eq(bt_conn_get_dst(conn), &clients[i].addr)) {
+			return;
+		}
 	}
 
-	return 0;
+	/* Else add the device */
+	for (uint8_t i = 0; i < ARRAY_SIZE(clients); i++) {
+		if (!atomic_test_bit(clients[i].flags, FLAG_ACTIVE)) {
+			atomic_set_bit(clients[i].flags, FLAG_ACTIVE);
+			memcpy(&clients[i].addr, bt_conn_get_dst(conn), sizeof(bt_addr_le_t));
+
+			/* Send out all pending notifications */
+			k_work_submit(&deferred_nfy_work);
+			return;
+		}
+	}
 }
 
+static void pacs_bond_deleted(uint8_t id, const bt_addr_le_t *peer)
+{
+	/* Find the device entry to delete */
+	for (int i = 0; i < ARRAY_SIZE(clients); i++) {
+		/* Check if match, and if active, if so, reset */
+		if (atomic_test_bit(clients[i].flags, FLAG_ACTIVE) &&
+		    bt_addr_le_eq(peer, &clients[i].addr)) {
+			for (uint8_t j = 0; j < FLAG_NUM; j++) {
+				atomic_clear_bit(clients[i].flags, j);
+			}
+			(void)memset(&clients[i].addr, 0, sizeof(bt_addr_le_t));
+			return;
+		}
+	}
+}
+
+static void pacs_security_changed(struct bt_conn *conn, bt_security_t level,
+				  enum bt_security_err err)
+{
+	LOG_DBG("%s changed security level to %d", bt_addr_le_str(bt_conn_get_dst(conn)), level);
+
+	if (err != 0 || conn->encrypt == 0) {
+		LOG_DBG("1");
+		return;
+	}
+
+	if (!bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+		LOG_DBG("2");
+		return;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(clients); i++) {
+
+		for (uint8_t j = 0; j < FLAG_NUM; j++) {
+			if (atomic_test_bit(clients[i].flags, j)) {
+
+				/**
+				 *  It's enough that one flag is set, as the defer work will go
+				 * through all notifiable characteristics
+				 */
+				k_work_submit(&deferred_nfy_work);
+				return;
+			}
+		}
+	}
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.security_changed = pacs_security_changed,
+};
+
+static struct bt_conn_auth_info_cb auth_callbacks = {
+	.pairing_complete = pacs_auth_pairing_complete,
+	.bond_deleted = pacs_bond_deleted
```

Comment:
Correct me if I am wrong, but I don't think `bond_delete` is called if a bond has been overwritten if `CONFIG_BT_KEYS_OVERWRITE_OLDEST=y`. 

You could argue that it should, but I don't think it currently is. I'll try to create a GH issue for that

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/60239

Reply:
Is there any actions that should be taken to circumvent this feature/bug?

Reply:
Seems there isn't any issue with this anymore